### PR TITLE
refactor(openapi)!: convert bundling library to openapi-cli

### PR DESCRIPTION
**Pull request by kleyow**

### CODE REVIEW

File: fspiop/v1_0/openapi3/components/schemas/Party.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+title: Party
+type: object
+description: Data model for the complex type Party.
+properties:
+  partyIdInfo:
+    $ref: ./PartyIdInfo.yaml
+    description: 'Party Id type, id, sub ID or type, and FSP Id.'
+  merchantClassificationCode:
```
Comment (eoln) [inline fspiop/v1_0/openapi3/components/schemas/Party.yaml:8]: in other places, merchantClassifiactionCode is restricted to have only four digits...

File: fspiop/v1_0/openapi3/components/schemas/PartyComplexName.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+title: PartyComplexName
+type: object
+description: Data model for the complex type PartyComplexName.
+properties:
+  firstName:
+    type: string
```
Comment (eoln) [inline fspiop/v1_0/openapi3/components/schemas/PartyComplexName.yaml:6]: what about the FirstName type defined above? - it has quite a restricted regexp pattern and here is allowed an unrestricted string

File: fspiop/v1_0/openapi3/components/schemas/QuotesIDPutResponse.yaml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+type: object
+description: 'The object sent in the PUT /quotes/{ID} callback.'
+properties:
+  transferAmount:
+    properties:
+      currency:
+        type: string
+        description: Currency of the amount.
+        example: USD
+      amount:
```
Comment (eoln) [inline fspiop/v1_0/openapi3/components/schemas/QuotesIDPutResponse.yaml:11]: what about using Amount type defined above?

File: fspiop/v1_0/openapi3/components/schemas/Refund.yaml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+title: Refund
+type: object
+description: Data model for the complex type Refund.
+properties:
+  originalTransactionId:
+    type: string
+    description: Reference to the original transaction ID that is requested to be refunded.
+    example: b51ec534-ee48-4575-b6a9-ead2955b8069
```
Comment (eoln) [inline fspiop/v1_0/openapi3/components/schemas/Refund.yaml:8]: as an example we are using the CorrelationId value (uuid v4)- but this type isn't used here - strange inconsequency...

File: fspiop/v1_0/openapi3/components/schemas/TransactionType.yaml [hunk 5]
```diff
@@ -0,0 +1,33 @@
+title: TransactionType
+type: object
+description: Data model for the complex type TransactionType.
+properties:
+  scenario:
+    type: string
+    description: 'Deposit, withdrawal, refund, …'
+    example: DEPOSIT
```
Comment (eoln) [inline fspiop/v1_0/openapi3/components/schemas/TransactionType.yaml:8]: why there is no enum guard here...

File: fspiop/v1_0/openapi3/components/schemas/TransactionsIDPutResponse.yaml [hunk 6]
```diff
@@ -0,0 +1,23 @@
+title: TransactionsIDPutResponse
+type: object
+description: 'The object sent in the PUT /transactions/{ID} callback.'
+properties:
+  completedTimestamp:
+    type: string
```
Comment (eoln) [inline fspiop/v1_0/openapi3/components/schemas/TransactionsIDPutResponse.yaml:6]: no regexp to ensure we have timestamp string?

File: fspiop/v1_0/openapi3/components/schemas/TransactionsIDPutResponse.yaml [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    description: Time and date when the transaction was completed.
+    example: '2016-05-24T08:38:08.699-04:00'
+  transactionState:
+    type: string
```
Comment (eoln) [inline fspiop/v1_0/openapi3/components/schemas/TransactionsIDPutResponse.yaml:10]: no enum here?

File: fspiop/v1_0/openapi3/openapi.yaml [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+servers:
+  - url: 'protocol://hostname:<port>/switch/'
+    variables:
+      protocol:
+        enum:
+          - http
+          - https
+        default: https
+paths:
+  /interface:
```
Comment (eoln) [inline fspiop/v1_0/openapi3/openapi.yaml:26]: nice workaround :-D

File: fspiop/v1_0/openapi3/paths/bulkTransfers_ID_error.yaml [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+  description: >-
+    If the server is unable to find or create a bulk transfer, or another
+    processing error occurs, the error callback `PUT /bulkTransfers/{ID}/error`
+    is used. The `{ID}` in the URI should contain the `bulkTransferId` that was
+    used for the creation of the bulk transfer (`POST /bulkTransfers`), or the
+    `{ID}` that was used in the `GET /bulkTransfers/{ID}` request.
+  summary: Return bulk transfer information error
+  tags:
+    - bulkTransfers
+  operationId: BulkTransfersErrorByID
```
Comment (eoln) [inline fspiop/v1_0/openapi3/paths/bulkTransfers_ID_error.yaml:11]: there is some inconsistency in naming operations, especially errors: `BulkTransfersErrorByID` vs `AuthorizationsByIDAndError`

File: fspiop/v1_0/openapi3/paths/participants.yaml [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+  description: >-
+    The HTTP request `POST /participants` is used to create information in the
+    server regarding the provided list of identities. This request should be
+    used for bulk creation of FSP information for more than one Party. The
+    optional currency parameter should indicate that each provided Party
+    supports the currency.
+  summary: Create bulk participant information
+  tags:
+    - participants
+  operationId: Participants1
```
Comment (eoln) [inline fspiop/v1_0/openapi3/paths/participants.yaml:11]: nit: extra `1` ?

File: fspiop/v1_0/openapi3/paths/participants_Type_ID.yaml [hunk 11]
```diff
@@ -0,0 +1,166 @@
...
+    The callback `PUT /participants/{Type}/{ID}` (or `PUT
+    /participants/{Type}/{ID}/{SubId}`) is used to inform the client of a
+    successful result of the lookup, creation, or deletion of the FSP
+    information related to the Party. If the FSP information is deleted, the
+    fspId element should be empty; otherwise the element should include the FSP
+    information for the Party.
+  summary: Return participant information
+  tags:
+    - participants
+  operationId: ParticipantsByTypeAndID3
```
Comment (eoln) [inline fspiop/v1_0/openapi3/paths/participants_Type_ID.yaml:99]: what this extra `3` is doing at the end?

File: fspiop/v1_0/openapi3/paths/parties_Type_ID_SubId.yaml [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+      $ref: ../components/responses/503.yaml
+put:
+  description: >-
+    The callback `PUT /parties/{Type}/{ID}` (or `PUT
+    /parties/{Type}/{ID}/{SubId}`) is used to inform the client of a successful
+    result of the Party information lookup.
+  summary: Return party information
+  tags:
+    - parties
+  operationId: PartiesSubIdByTypeAndIDPut
```
Comment (eoln) [inline fspiop/v1_0/openapi3/paths/parties_Type_ID_SubId.yaml:54]: sometimes we have `magic` number at the end elsewhere the name of `put` http method and in the same file we don't have the same naming strategy for `get`

File: fspiop/v1_1/openapi3/components/headers/Content-Type.yaml [hunk 13]
```diff
@@ -0,0 +1,6 @@
+schema:
+  type: string
```
Comment (eoln) [inline fspiop/v1_1/openapi3/components/headers/Content-Type.yaml:2]: unrestricted string, some pattern here would be great!

File: fspiop/v1_1/openapi3/components/parameters/Accept.yaml [hunk 14]
```diff
@@ -3,6 +3,6 @@ in: header
 required: true
 schema:
   type: string
-description: |
-  The `Accept` header field indicates the version of the API the client
-  would like the server to use.
+description: >-
```
Comment (eoln) [inline fspiop/v1_1/openapi3/components/parameters/Accept.yaml:6]: again unrestricted string where some regexp pattern would place here nicely

File: fspiop/v1_1/openapi3/components/parameters/Date.yaml [hunk 15]
```diff
@@ -0,0 +1,6 @@
+name: Date
+in: header
+schema:
+  type: string
```
Comment (eoln) [inline fspiop/v1_1/openapi3/components/parameters/Date.yaml:4]: Date should be guarded by regexp

File: fspiop/v1_1/openapi3/components/parameters/FSPIOP-HTTP-Method.yaml [hunk 16]
```diff
@@ -0,0 +1,11 @@
+name: FSPIOP-HTTP-Method
+in: header
+schema:
+  type: string
```
Comment (eoln) [inline fspiop/v1_1/openapi3/components/parameters/FSPIOP-HTTP-Method.yaml:4]: why not restricted string where a pattern or enum should be applied

File: fspiop/v1_1/openapi3/components/parameters/FSPIOP-Signature.yaml [hunk 17]
```diff
@@ -3,6 +3,6 @@ in: header
 schema:
   type: string
 required: false
-description: |
+description: >-
```
Comment (eoln) [inline fspiop/v1_1/openapi3/components/parameters/FSPIOP-Signature.yaml:6]: signatures are base64 encoded string so we can guard this by regexp pattern...

**Review by eoln — APPROVED** [review]
@kleyow GJ! Approved

I put a lot of comments related to some inconsistency and lack of regexp patterns & enums inherited from the original specification. 
 
@elnyry-sam-k @lewisdaly we should decide what to do about -> should we create issue and delegate complaints of very poor API spec quality to some board? 

Or maybe we should fix these problems in api-snippets - if yes then definitely in other story...

Worth mentioning that api-snippets restructuring allows spotting these imperfections much easier than reading original spec doc...

File: fspiop/v1_0/openapi3/components/schemas/Amount.yaml [hunk 18]
```diff
@@ -0,0 +1,10 @@
+title: Amount
+type: string
+pattern: '^([0]|([1-9][0-9]{0,17}))([.][0-9]{0,3}[1-9])?$'
```
[the remark this answers is not in the document]
Comment (elnyry-sam-k) [reply fspiop/v1_0/openapi3/components/schemas/Amount.yaml:3]: This is by design, why is it wrong? isn't that how match works? zeroes at the end after decimals add `zero` value and not necessary :P isn't it?

File: fspiop/v1_0/openapi3/components/schemas/Party.yaml (hunk 1 above)
Comment (elnyry-sam-k) [reply fspiop/v1_0/openapi3/components/schemas/Party.yaml:8]: yes merchantClassifiactionCode is different from FSP ID, sorry I'm not understanding the issue with that

**Comment by elnyry-sam-k** [comment]
@eoln - great work reviewing the giant PR :-).. on the issues / enhancements you mentioned as comments, can we take those separately and not include them here ( for example, regex patterns on headers, etc)

**Comment by eoln** [comment]
@elnyry-sam-k that is my idea, move improvements to another story, we are chopping big spec here only

File: README.md [hunk 19]
```diff
@@ -13,20 +13,20 @@ Install the snippet library
...
+  npm install @redocly/openapi-cli --save-dev
 ```
 
 Modify swagger file to reference `api-snippets`.
 
 ex.
 ```yaml
 Money:
-  $ref: /path/to/node_modules/@mojaloop/api-snippets/v1.0/openapi3/schemas/Money.yaml
+  $ref: /path/to/node_modules/@mojaloop/api-snippets/fspiop/v1_1/openapi3/components/schemas/Money.yaml
```
[the remark this answers is not in the document]
Comment (kleyow) [reply README.md:24]: Yes I will make a ticket to investigate this.

**Pull request merged**